packages/yuudachi/src/commands/moderation/sub/report/message.ts

```diff
@@ -102,36 +109,68 @@ export async function message(
 		});
 	} else if (collectedInteraction?.customId === reportKey) {
 		await collectedInteraction.deferUpdate();
+		try {
+			if ((await redis.smembers(messageKey)).includes(args.message.id)) {
+				await collectedInteraction.editReply({
+					content: i18next.t("command.utility.report.common.errors.recently_reported.message", { lng: locale }),
+					embeds: [],
+					components: [],
+				});
 
-		if (await redis.exists(key)) {
-			await collectedInteraction.editReply({
-				content: i18next.t("command.utility.report.common.errors.recently_reported.message", { lng: locale }),
-				embeds: [],
-				components: [],
-			});
+				return;
+			}
 
-			return;
-		}
+			await redis.sadd(messageKey, args.message.id);
+			if (userRecentlyReported) {
+				try {
+					await forwardReport(
+						{
+							author: collectedInteraction.user,
+							reason: trimmedReason,
+						},
+						args.message as Message<true>,
+						locale,
+					);
+				} catch (error) {
+					await collectedInteraction.editReply({
+						content: (error as Error).message,
+						components: [],
+						embeds: [],
+					});
+					return;
+				}
```

Comment:
Shouldn't we do this check in `validateReport` in `moderation/report.ts` instead, so the user doesn't have to fill out a modal for example and go through all the trouble just to get an error message?

Reply:
Changed in 78edd51